runtime/runtime/src/actions.rs

```diff
@@ -60,6 +60,11 @@ pub(crate) fn action_function_call(
     is_last_action: bool,
     epoch_info_provider: &dyn EpochInfoProvider,
 ) -> Result<(), RuntimeError> {
+    // TODO: maybe we don't need it in such a way.
+    let cache = match &apply_state.cache {
```

Comment:
Can you simply do `let cache = apply_state.cache.as_ref()`?

Reply:
Not this way, we need to convert Arc'ed value to reference. This code will generate `Option<Arc<>>`.